src/openzaak/components/catalogi/admin/admin_views.py

```diff
@@ -167,4 +177,25 @@ def post(self, request, *args, **kwargs):
             return HttpResponseRedirect(reverse("admin:catalogi_catalogus_changelist"))
         except (CommandError, IntegrityError) as exc:
             messages.add_message(request, messages.ERROR, exc)
+
+        catalogus_pk = kwargs.get("catalogus_pk")
+        catalogus = Catalogus.objects.get(pk=catalogus_pk)
+
+        if iotype_forms:
+            for i, form in zip(request.session["iotypen"], iotype_forms.forms):
+                form._bound_fields_cache["existing"].label = (
+                    str(catalogus) + " - " + i["omschrijving"]
+                )
+            kwargs["iotype_forms"] = iotype_forms
```

Comment:
This feels kinda hacky but otherwise it I would need to iterative over the labels above. Not sure if this actually improves performance.

Reply:
All the fields default to `Existing` if you do not manually set the labels after getting the form from post. 

I places this at the end of the code since you only need the labels if the code fails. It might make more sense just to initialize the labels regardless.

Reply:
Why can't it be solved by putting formsets into the context?

Reply:
Ok I managed to reproduce it.
But you create formsets in GET and in POST in different ways - in POST call you don't have `form_kwargs` parameter, which has labels. If you add `form_kwargs` when initializing formsets in POST the labels would be correct
At least it works on my machine